### PR TITLE
expose the departing unit on RelationDeparted events

### LINKED ISSUE #707: Exposure of `JUJU_DEPARTING_UNIT`
Hi team! Would it be possible to expose `JUJU_DEPARTING_UNIT`? Right now I am accessing it via `os.environ`. 

The motivation is that when a unit is getting removed I don't want the unit that is departing from the application to set any reconfigurations. A simple check for if the current unit is the departing unit seems to solve this. 😄

### PR DESCRIPTION
**Pull request by rwcarlsen**
I refactored a bit of the _get_event_args func logic.  I'd appreciate
some review focus on that to make sure I didn't subtley change/break
behavior.

Fixes #707

### CODE REVIEW

**Comment by rwcarlsen** [comment]
@MiaAltieri @balbirthomas

**Review by pengale — APPROVED** [review]
This is in keeping with the way that things are currently setup, and I am +1 on this PR.

That said, ops/main.py is a mess, and we need to fix it at some point. I would like to replace most of the branching logic in its various methods with a HookEnv struct that gets all the possible environment variables, and puts them in a place that all the HookEvent classes can access.

That would mean that things like RelationEvent would be able to drop most of the complicated args they take in their init, and just look up stuff in the HookEnv as needed.

But that's for later. For now, this is good :-)

File: ops/charm.py [hunk 1]
```diff
@@ -388,7 +388,8 @@ class RelationDepartedEvent(RelationEvent):
     This is the inverse of the :class:`RelationJoinedEvent`, representing when a
     unit is leaving the relation (the unit is being removed, the app is being
     removed, the relation is being removed). It is fired once for each unit that
-    is going away.
+    is going away for remaining units, and once for every remaining unit for the
```
Comment (balbirthomas) [inline ops/charm.py:391]: nitpick: "It is fired once for each unit that is going away for remaining units, and once for every remaining unit for the departing unit." is a bit troublesome to grok (maybe it is just me so feel free to disregard this suggestion). Would the following be accurate and acceptable "It is fired once for each unit that is going away. It is also fired for all the other units that remain".

Comment (rwcarlsen) [reply ops/charm.py]: Tweaked it - should be more clear now.

**Review by jnsgruk — APPROVED** [review]
Nice work on the refactoring of the get_event_args, much easier to read now. Change looks good to me.

File: ops/main.py [hunk 2]
```diff
@@ -172,22 +173,26 @@ def _get_event_args(charm, bound_event):
...
-            if '/' not in remote_unit_name:
-                raise RuntimeError('invalid remote unit name: {}'.format(remote_unit_name))
-            remote_app_name = remote_unit_name.split('/')[0]
-        args = [relation, model.get_app(remote_app_name)]
-        if remote_unit_name:
-            args.append(model.get_unit(remote_unit_name))
-        return args, {}
-    elif relation:
-        return [relation], {}
+    departing_unit_name = os.environ.get('JUJU_DEPATING_UNIT', '')
```
Comment (PietroPasotti) [inline ops/main.py:179]: DEPATING unit?

**Pull request merged**